### PR TITLE
feat: add fantom implementation

### PR DESCRIPTION
**Pull request by kemumaki375**
Here come Fantom implementation which nobody ask for. Code based on vanilla implementation and Fantom's [dom example](https://fantom.org/doc/examples/js-dom).

### CODE REVIEW

**Comment by herlangga72** [comment]
Can you explain how it works?

Is that from java code then compile to native js or what??

**Comment by kemumaki375** [comment]
> **Can you explain how it works?**

[Fantom](https://fantom.org/) is a programming language that can target JVM, Javascript, and .NET. There are two files that you need to be aware:
- main.fan: containing code that handle web request and generating html response.
- script.fan: containing code that will be compiled on the fly to native Javascript. This file will be referenced from main.fan

I'm including the resulting static file on the `dist` folder which can be run directly using web browser. 

> **Is that from java code then compile to native js or what??**

No and Yes. No, fantom is a separate programming language that can target JVM and Javascript. Yes, it can be compiled to Javascript at runtime. It can serve both http request and also generate client side script from single project.
 
> **Is there any method to make it just frontend only?**

I'm not aware as of now since the html part are generated directly from the server using Fantom specifics [library ](https://fantom.org/doc/docDomkit/index) and syntax. I will find out if it possible. 

Please note that I'm including the html and javascript file in the 'dist' folder which can be run directly from the browser or static web server. It's not automatically generated though (I save it manually from the browser).

**Comment by kemumaki375** [comment]
Sorry for the mess (closing and reopening the pr), accidentally push my other commit to the wrong branch and struggle a little bit when fixing it 😌✌️

**Review by herlangga72 — APPROVED** [review]
LGTM

File: fantom/readme.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Readme
+[Fantom](https://www.fantom.org/) implementation of Teknum's decimal clock based on [vanilla](https://github.com/teknologi-umum/decimal-clock/tree/main/vanillay) implementation. This code need to be run on the server side. Content inside `dist` folder does not coming from build output, but rather from manually saving the generated page and script from browser.
```
[the remark this answers is not in the document]
Comment (kemumaki375) [reply fantom/readme.md:2]: If want to run using fantom source code directy - then yes, you need to run it on the server. The server mostly used to serve http request and generate html output. 

However, the logic behind the clock itself (which is also written in fantom) are compiled into native javascript.  it's like running python flask but the script part is also written using python.

**Review by ronnygunawan — APPROVED** [review]
Interesting language

**Pull request merged**